File: algorithms/c/169.c
#include <stdio.h>

int majorityElement(int* nums, int numsSize)
{
    int i;
    int ans, count;
    ans = nums[0], count = 1;
    for ( i = 1; i < numsSize; i++ )
    {
        if ( count == 0 )
            ans = nums[i];
        if ( nums[i] == ans )
            count++;
        else
            count--;
    }
    return ans;
}

int main()
{
    return 0;
}

File: algorithms/python/169.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ans = nums[0]
        count = 1
        for i in range(1, len(nums)):
            if count == 0:
                ans = nums[i]
            if nums[i] == ans:
                count += 1
            else:
                count -= 1
        return ans
